### PR TITLE
chop over 200mb off the Docker container

### LINKED ISSUE #425: Make our container slimmer
```
 ~ $ docker history mozillamarketplace/solitude:latest
IMAGE               CREATED             CREATED BY                                      SIZE
f50827b510a3        5 hours ago         /bin/sh -c python /srv/solitude/bin/crontab/g   798 B
168ad93f67cc        5 hours ago         /bin/sh -c #(nop) ENV HISTCONTROL=erasedups     0 B
2172047d7d84        5 hours ago         /bin/sh -c #(nop) ENV HISTIGNORE=ls:exit:cd .   0 B
5db77155ecfd        5 hours ago         /bin/sh -c #(nop) ENV HISTSIZE=50000            0 B
f72763ab846a        5 hours ago         /bin/sh -c #(nop) ENV HISTFILE=/srv/solitude/   0 B
3b901754e6d6        5 hours ago         /bin/sh -c #(nop) EXPOSE 2602/tcp               0 B
19a31905244f        5 hours ago         /bin/sh -c #(nop) ENV SOLITUDE_DATABASE=mysql   0 B
3ae2ce2ec2b1        5 hours ago         /bin/sh -c cd /srv/solitude && git show -s --   7 B
e3bd6d47cba3        5 hours ago         /bin/sh -c #(nop) COPY dir:80a9700e42be3881fc   1.475 MB
e34d72146094        5 hours ago         /bin/sh -c peep install     --build /pip/buil   72.67 MB
f1aaacc03f8e        5 hours ago         /bin/sh -c pip install --no-deps --find-links   87.54 kB
1b4b8c84d479        5 hours ago         /bin/sh -c #(nop) WORKDIR /pip                  0 B
e3dea5d56be5        5 hours ago         /bin/sh -c #(nop) ADD dir:a3401d8869880d2d5ff   7.48 kB
5d1269ac6ea1        5 hours ago         /bin/sh -c mkdir -p /pip/{cache,build}          0 B
360e529bb331        5 hours ago         /bin/sh -c #(nop) ENV IS_DOCKER=1               0 B
acfb7a83f98c        5 hours ago         /bin/sh -c yum install -y cronie                34.72 MB
5d099af9ad32        5 hours ago         /bin/sh -c yum install -y bash-completion       18.68 MB
16bb922e5ff5        5 hours ago         /bin/sh -c yum install -y supervisor            224.5 MB
a921f723427c        8 months ago        /bin/sh -c yum install -y mysql-community-ser   375 MB
8ad265df9720        8 months ago        /bin/sh -c #(nop) ADD file:aef05b6e4bc0dc6ad7   144 B
e8bea049a4f1        8 months ago        /bin/sh -c pip install ipython ipdb             12.85 MB
36ffb00bde49        8 months ago        /bin/sh -c easy_install pip                     5.37 MB
cf917f76bec1        8 months ago        /bin/sh -c #(nop) ENV LD_LIBRARY_PATH=/opt/rh   0 B
03afde2a3cea        8 months ago        /bin/sh -c #(nop) ENV PATH=/opt/rh/python27/r   0 B
ceaa1c83f483        8 months ago        /bin/sh -c yum install -y libxslt               18.13 MB
c0c274ab3c92        8 months ago        /bin/sh -c yum install -y     python27-m2cryp   24.07 MB
c7231ca07c09        8 months ago        /bin/sh -c yum install -y gcc                   52.66 MB
9d6465866850        8 months ago        /bin/sh -c yum install -y git                   180.4 MB
01c25c4a6066        8 months ago        /bin/sh -c yum install -y python27              178 MB
42fbabc14120        8 months ago        /bin/sh -c #(nop) ADD file:6f87960d9923b0cb2b   123 B
68eb857ffb51        8 months ago        /bin/sh -c #(nop) ADD file:9be0e96ac5792a45d4   212.7 MB
34e94e67e63a        11 months ago       /bin/sh -c #(nop) MAINTAINER The CentOS Proje   0 B
511136ea3c5a        23 months ago                                                       0 B
```

git, python, supervisor and mysql-community-server are the biggest culprits

### PR DESCRIPTION
**Pull request by andymckay**
helps fix #425, most of the rest of the fixes are in the base container

### CODE REVIEW

**Comment by andymckay** [comment]
Because we now copy the source into the container, there's no need to do anything seperate with pip.

File: Dockerfile [hunk 1]
```diff
@@ -4,30 +4,24 @@
...
+# if it wants to.
+COPY . /srv/solitude
+RUN cd /srv/solitude && git show -s --pretty="format:%h" > git-rev.txt
+
 # Download this securely from pyprepo first.
 RUN pip install --no-deps --find-links https://pyrepo.addons.mozilla.org/ peep
 RUN peep install \
-    --build /pip/build \
-    --download-cache /pip/cache \
+    --download-cache /tmp/pip/cache \
```
Comment (kumar303) [inline Dockerfile]: Do you need to ``mkdir -p /tmp/pip/cache`` ? Peep doesn't actually use the cache dir so you can probably remove it for now. https://github.com/erikrose/peep/issues/93

**Comment by kumar303** [comment]
This is pretty surprising! I guess removing commands removes layers in mysterious ways?

r+wc from me

File: Dockerfile (hunk 1 above)
Comment (andymckay) [inline Dockerfile]: No pip creates the directory if it doesn't exist. If peep doesn't use it, its gone... wasn't quite sure of the point in this context anyway.

**Pull request merged**